### ISSUE TITLE
MouseSystem detect transparency

### ISSUE THREAD

**Issue by EtienneBruines** [EngoEngine/engo#227]
It would be nice if the MouseSystem detected transparency, and had an option to let it "not fire" an event whenever you hover the transparent background. 

Use-case: you have a red circle with transparent background. On hovering, you want the cursor to change. You want the cursor to change *only* whenever you hover the red part, not the transparent background.

**Comment by otraore** [comment]
I don't think this should be the game engine's responsibility rather than the user. If you have a circle regardless of what's inside of it, when the mouse hovers above it it should fire the mouse hover event. If you have a circle object there should be a way to define where it collides as well (edges).

**Comment by EtienneBruines** [comment]
We should be able to handle circular objects in some way or the other. The `MouseSystem` has an "easy" fix in that you can detect the transparency-level of the current pixel within the texture you're hovering. Not sure how fail-proof that is with rotation in place. The `CollisionSystem` would indeed be even harder. 

But when we offer a `MouseSystem` or `CollisionSystem` (which we do), we should at least try to make them work with circular objects (or at least leave this issue open because we don't know how to fix it).

**Comment by otraore** [comment]
@EtienneBruines I understand but you are taking some extensibility from the user whereas they could handle it themselves. How about in the event somehow let them now if the collision was a transparent hover, I haven't looked into the messaging/event system that much so I'm not sure how yet.

**Comment by EtienneBruines** [comment]
Well, the "event firing" is currently a matter of setting `entity.MouseComponent.Hover` to `true`. 

We could optionally allow them to set `CheckForTransparency` or something when adding the `MouseSystem`, which would be more strict than the current implementation.

**Comment by otraore** [comment]
Sure @EtienneBruines sounds good

**Comment by EtienneBruines** [comment]
Hmm, everything breaks with rotation (#216), this is gonna need some more thought than I hoped...

**Comment by paked** [comment]
@EtienneBruines Wouldn't rotation with circles be easier lol?

**Comment by EtienneBruines** [comment]
@paked Lol, yes, if you rotate around the center. Which we don't do yet.
